### DISCUSSION TITLE
How to layout these classes to avoid inheritance information overlapped together

### DISCUSSION THREAD

**Discussion by Akarinnnnn** [plantuml/plantuml#1722] [Q&A] [answered]
I have a `CompositeLayer` class that inherits 4 interfaces which inherits another generic interface, and I'm drawing a class diagram about it. Here is the plantuml code:
![code](https://github.com/plantuml/plantuml/assets/43724908/62cff4f1-71f0-454c-a15d-f86dd1c7cddd)

And I get such outcome through plantuml-1.2024.3:
![outcome, specifiers stack together](https://github.com/plantuml/plantuml/assets/43724908/e46e2375-5813-4765-90ec-d8d9edd0ec11)

Noticed that generic inheritance information of keyboard and gamepad handler overlapped together, but I expect they don't overlap. Is there any way to achieve this?

**Comment by The-Lum** [comment]
Hi,

You can use `\n` to add some more spaces...

_If that can help,_
Regards,
Th.

**Comment by The-Lum** [comment, marked as answer]
Hi @Akarinnnnn, and all,

Here is a minimal example, from:
```puml
a "a2b long long long lib" <|-- b
a "a2c long long long lib" <|-- c
a "a2d long long long lib" <|-- d
a "a2z long long long lib" <|-right- z
```
>[![](https://www.planttext.com/api/plantuml/svg/IrHGIZHAKiZ9pqj79ZAJb1HiQdHr5PAu4a4gad6hI8QeIC6j8WMYeWgNYgBCz8mINOKg0000)](https://www.planttext.com/?text=IrHGIZHAKiZ9pqj79ZAJb1HiQdHr5PAu4a4gad6hI8QeIC6j8WMYeWgNYgBCz8mINOKg0000)
_[FYI @arnaudroques for the overlapping issue (observed on v1.2024.4beta6)]_

1/ Use `\n`
```puml
a "a2b long long long lib" <|-- b
a "\na2c long long long lib" <|-- c
a "\n\na2d long long long lib" <|-- d
a "a2z long long long lib" <|-right- z
```
>[![](https://www.planttext.com/api/plantuml/svg/IrHGIZHAKiZ9pqj79ZAJb1HiQdHr5PAu4XMKOl8IZP9ngqc6g06fIi6jAWMaAj6e2fUAeipqZ19TXIe0)](https://www.planttext.com/?text=IrHGIZHAKiZ9pqj79ZAJb1HiQdHr5PAu4XMKOl8IZP9ngqc6g06fIi6jAWMaAj6e2fUAeipqZ19TXIe0)


2/ Or Change to arrow labels as:
```puml
a <|-- b       : "a2b long long long lib"
a <|-- c       : "a2c long long long lib"
a <|-- d       : "a2d long long long lib"
a <|-right- z  : "a2z long long long lib"
```
>[![](https://www.planttext.com/api/plantuml/svg/IrImgT7LLKXIW00h1QL4eoI5dFoyT6GYCqc9An6YCXbPPJ8-bId8AbDmgopAJCyeqLMeWggimgOI0000)](https://www.planttext.com/?text=IrImgT7LLKXIW00h1QL4eoI5dFoyT6GYCqc9An6YCXbPPJ8-bId8AbDmgopAJCyeqLMeWggimgOI0000)

_If that can help,_
Regards,
Th.

**Comment by Akarinnnnn** [comment]
Thank you for providing help, I'll try later!



---Original---
From: ***@***.***&gt;
Date: Wed, Apr 3, 2024 04:17 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [plantuml/plantuml] How to layout these classes to avoidinheritance information overlapped together (Discussion #1722)




 
Hi @Akarinnnnn, and all,
 
Here is a minimal example, from:
 a "a2b long long long lib" <|-- b a "a2c long long long lib" <|-- c a "a2d long long long lib" <|-- d a "a2z long long long lib" <|-right- z
  

 [FYI @arnaudroques for the overlapping issue (observed on v1.2024.4beta6)]
  
1/ Use \n
 a "a2b long long long lib" <|-- b a "\na2c long long long lib" <|-- c a "\n\na2d long long long lib" <|-- d a "a2z long long long lib" <|-right- z
  

  
2/ Or Change to arrow labels as:
 a <|-- b       : "a2b long long long lib" a <|-- c       : "a2c long long long lib" a <|-- d       : "a2d long long long lib" a <|-right- z  : "a2z long long long lib"
  

  
If that can help,
 Regards,
 Th.

**Reply by VladimirAlexiev** [reply]
@Akarinnnnn to clarify "2/ Or Change to arrow labels":
Your syntax as per "1/" should only be used for cardinalities (short labels that are placed near the head/tail of the arrow.

And I can add an ugly variant:

3/ Use arrow lengths to shuffle the arrows to different positions, eg `<|------` is a longish arrow.
